soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java

```diff
@@ -0,0 +1,271 @@
+import org.apache.curator.test.TestingServer;
+import org.dromara.soul.common.constant.ZkPathConstants;
+import org.dromara.soul.common.dto.AppAuthData;
+import org.dromara.soul.common.dto.ConditionData;
+import org.dromara.soul.common.dto.MetaData;
+import org.dromara.soul.common.dto.PluginData;
+import org.dromara.soul.common.dto.RuleData;
+import org.dromara.soul.common.dto.SelectorData;
+import org.dromara.soul.common.enums.MatchModeEnum;
+import org.dromara.soul.common.enums.OperatorEnum;
+import org.dromara.soul.common.enums.ParamTypeEnum;
+import org.dromara.soul.common.enums.PluginEnum;
+import org.dromara.soul.common.enums.SelectorTypeEnum;
+import org.dromara.soul.sync.data.api.AuthDataSubscriber;
+import org.dromara.soul.sync.data.api.MetaDataSubscriber;
+import org.dromara.soul.sync.data.api.PluginDataSubscriber;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Test cases for ZookeeperSyncDataService.
+ *
+ * @author zendwang
+ */
+@Slf4j
+@SuppressWarnings("all")
+public final class ZookeeperSyncDataServiceTest {
+    
+    private static final Map<String, PluginData> PLUGIN_ZK_DTO_MAP = Maps.newConcurrentMap();
+    
+    private static TestingServer zkServer;
+    
+    private ZkClient zkClient;
+    
+    private ZookeeperSyncDataService syncDataService;
+    
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        zkServer = new TestingServer(21810, true);
+    }
+    
+    @Before
+    public void setUp() throws Exception {
+        zkClient = new ZkClient("127.0.0.1:21810");
+        buildZkData();
+        PluginDataSubscriber pluginDataSubscriber = Mockito.mock(PluginDataSubscriber.class);
+        MetaDataSubscriber metaDataSubscriber = Mockito.mock(MetaDataSubscriber.class);
+        List<MetaDataSubscriber> metaDataSubscribers = Lists.newArrayList(metaDataSubscriber);
+        AuthDataSubscriber authDataSubscriber = Mockito.mock(AuthDataSubscriber.class);
+        List<AuthDataSubscriber> authDataSubscribers = Lists.newArrayList(authDataSubscriber);
+        syncDataService = new ZookeeperSyncDataService(zkClient, pluginDataSubscriber, metaDataSubscribers, authDataSubscribers);
+    }
+    
+    @After
+    public void after() {
+        zkClient.close();
+        syncDataService.close();
+    }
+    
+    @AfterClass
+    public static void tearDown() throws Exception {
+        zkServer.stop();
+    }
+    
+    @SneakyThrows
+    @Test
+    public void testWatcherPlugin() {
+        PluginData pluginData = new PluginData("6", PluginEnum.DIVIDE.getName(), "aaaaa", 0, Boolean.TRUE);
+        writePlugin(pluginData);
+        final String pluginPath = ZkPathConstants.buildPluginPath(pluginData.getName());
```

Comment:
Can you also test `onSubscribe ` method? I think we should verify the data from watch. similar in other places. Thanks.

Reply:
ok,I will do it